examples/minimal/client/index.ts

```diff
@@ -1,19 +1,33 @@
 import { createTRPCProxyClient, httpBatchLink } from '@trpc/client';
 import type { AppRouter } from '../server';
 
-const client = createTRPCProxyClient<AppRouter>({
+// Add polyfill for fetch for Node < 18. (Stackblitz runs Node 16 and we want embed to work)
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+global.fetch = require('undici').fetch;
+
+const trpc = createTRPCProxyClient<AppRouter>({
   links: [
     httpBatchLink({
-      url: 'http://localhost:2022',
+      url: 'http://localhost:3000',
     }),
   ],
 });
 
 async function main() {
-  const result = await client.greet.query('tRPC');
+  /**
+   * Inferring types
+   */
+  const users = await trpc.userList.query();
+  //    ^?
+  console.log('Users:', users);
+
+  const createdUser = await trpc.userCreate.mutate({ name: 'sachinraja' });
+  //    ^?
+  console.log('Created user:', createdUser);
 
-  // Type safe
-  console.log(result.greeting.toUpperCase());
+  const user = await trpc.userById.query('1');
```

Comment:
This is still broken afaik. It'll report `User`, despite the fact that it is in fact `User | undefined`.
It's only broken with `createTRPCProxyClient`, not `createCaller`, as default transform type is making the query output to be passed through `Serialize` type, which removes the `undefined` part.

Reply:
ref: https://github.com/trpc/trpc/issues/4049#issuecomment-1484868645